sql/memo/rel_props.go

```diff
@@ -327,14 +328,53 @@ func statsForRel(rel RelExpr) sql.Statistic {
 		// different joins use different ways to estimate cardinality of outputs
 		jp := rel.JoinPrivate()
 		left := jp.Left.RelProps.GetStats()
-		right := jp.Left.RelProps.GetStats()
+		right := jp.Right.RelProps.GetStats()
 
-		switch n := rel.(type) {
-		case *LookupJoin:
-			if n.Injective {
+		var injective bool
+		var mergeStats sql.Statistic
+		var n RelExpr = rel
+		var done bool
+		for n != nil && !done {
+			switch n := n.(type) {
+			case *LookupJoin:
+				injective = injective || n.Injective
+			case *MergeJoin:
```

Comment:
I think this section could use some more comments explaining why certain checks / assignments are done.